### PR TITLE
Adds Authentication related endpoints

### LINKED ISSUE #28: create user account routes / User Auth
parent: #16 
### routes
- [x] /users/auth/register
- [x] /users/auth/login

### Logic/Data changes
- [x] proper implementation for the UserDetailsService
- [x] add db table for users
- [x] add account logic handler
- [ ] #67

### Useful References
https://www.javaguides.net/2021/10/login-and-registration-rest-api-using-spring-boot-spring-security-hibernate-mysql-database.html
https://www.baeldung.com/get-user-in-spring-security
https://docs.spring.io/spring-security/reference/servlet/configuration/java.html#jc-httpsecurity

### PR DESCRIPTION
**Pull request by VanJackal**
closes #28

### CODE REVIEW

File: critterSpring/src/main/java/com/combatcritters/critterspring/auth/AuthController.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            return new ResponseEntity<>("Username already exists", HttpStatus.CONFLICT);
+        }
+        
+        //create the new user
+        User user = userManager.createUser(
+                payload.username(), 
+                passwordEncoder.encode(payload.password())
+        );
+        return new ResponseEntity<>("Registered new user:"+user.getUsername(), HttpStatus.CREATED);
+    }
```
Comment (VanJackal) [inline critterSpring/src/main/java/com/combatcritters/critterspring/auth/AuthController.java:71]: Both of these methods should have javadocs

**Comment by whatdoes3plus1equalsto** [comment]
Need to put UserManagerTest into AllUnitTests.java
apart from that, looks good

**Comment by VanJackal** [comment]
> Need to put UserManagerTest into AllUnitTests.java apart from that, looks good

AllUnitTests should probably be removed, gradle tasks `test` and `integrationTest` handle this.

**Review by whatdoes3plus1equalsto — APPROVED** [review]
all good

**Pull request merged**